test/e2e_node/split_disk_test.go

```diff
@@ -0,0 +1,309 @@
+			time.Sleep(30 * time.Second)
+			ginkgo.By("setting up pods to be used by tests")
+			pods := []*v1.Pod{}
+			for _, spec := range testSpecs {
+				pods = append(pods, spec.pod)
+			}
+			e2epod.NewPodClient(f).CreateBatch(ctx, pods)
+		})
+
+		ginkgo.It("should eventually evict all of the correct pods", func(ctx context.Context) {
+			_, is, err := getCRIClient()
+			framework.ExpectNoError(err)
+			resp, err := is.ImageFsInfo(ctx)
+			framework.ExpectNoError(err)
+			gomega.Expect(resp.ImageFilesystems).NotTo(gomega.BeEmpty())
+			gomega.Expect(resp.ImageFilesystems[0].FsId).NotTo(gomega.BeNil())
+			diskToPressure := filepath.Dir(resp.ImageFilesystems[0].FsId.Mountpoint)
+			ginkgo.By(fmt.Sprintf("Got imagefs directory: %s", diskToPressure))
+			imagesLenBeforeGC := 1
+			sizeOfPressure := "8000"
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				images, err := is.ListImages(ctx, &runtimeapi.ImageFilter{})
+				imagesLenBeforeGC = len(images)
+				return err
+			}, 1*time.Minute, evictionPollInterval).Should(gomega.Succeed())
+			ginkgo.By(fmt.Sprintf("Images found before GC was %d", imagesLenBeforeGC))
+			ginkgo.By(fmt.Sprintf("Induce disk pressure on %s with size %s", diskToPressure, sizeOfPressure))
+			gomega.Expect(runDDOnFilesystem(diskToPressure, sizeOfPressure)).Should(gomega.Succeed())
+			ginkgo.By(fmt.Sprintf("Waiting for node to have NodeCondition: %s", expectedNodeCondition))
+
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				logFunc(ctx)
+				if expectedNodeCondition == noPressure || hasNodeCondition(ctx, f, expectedNodeCondition) {
+					return nil
+				}
+				return fmt.Errorf("NodeCondition: %s not encountered", expectedNodeCondition)
+			}, pressureTimeout, evictionPollInterval).Should(gomega.BeNil())
+
+			ginkgo.By("Waiting for evictions to occur")
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				if expectedNodeCondition != noPressure {
+					if hasNodeCondition(ctx, f, expectedNodeCondition) {
+						framework.Logf("Node has %s", expectedNodeCondition)
+					} else {
+						framework.Logf("Node does NOT have %s", expectedNodeCondition)
+					}
+				}
+				logKubeletLatencyMetrics(ctx, kubeletmetrics.EvictionStatsAgeKey)
+				logFunc(ctx)
+				return verifyEvictionOrdering(ctx, f, testSpecs)
+			}, pressureTimeout, evictionPollInterval).Should(gomega.Succeed())
+
+			ginkgo.By("checking for the expected pod conditions for evicted pods")
+			verifyPodConditions(ctx, f, testSpecs)
+
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				images, err := is.ListImages(ctx, &runtimeapi.ImageFilter{})
+				if err != nil {
+					return err
+				}
+				imagesLenAfterGC := len(images)
+				if imagesLenAfterGC < imagesLenBeforeGC {
+					return nil
+				} else {
+					return fmt.Errorf("garbage collection of images should have occurred. before: %d after:%d", imagesLenBeforeGC, imagesLenAfterGC)
+				}
```

Comment:
Since these are both return statements, there is no need for else in the if-else statement.

```suggestion
if imagesLenAfterGC < imagesLenBeforeGC {
	return nil
}

return fmt.Errorf("garbage collection of images should have occurred. before: %d, after: %d", imagesLenBeforeGC, imagesLenAfterGC)
```